docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -0,0 +1,771 @@
+
+[source,properties]
+----
+%dev.quarkus.datasource.jdbc.url=jdbc:mysql://localhost:3306/mydatabase?useSSL=false
+quarkus.datasource.jdbc.url=jdbc:mysql://remotehost:3306/mydatabase?useSSL=false
+----
+
+can be simplified by having:
+
+[source,properties]
+----
+%dev.application.server=localhost
+application.server=remotehost
+
+quarkus.datasource.jdbc.url=jdbc:mysql://${application.server}:3306/mydatabase?useSSL=false
+----
+
+It does result in one more line in this example but the value of `application.server` can be reused in other properties,
+diminishing the possibility of typos and providing more flexibility in property definitions.
+
+
+[#combine-property-env-var]
+== Combining Property Expressions and Environment Variables
+
+Quarkus also supports the combination of both property expressions and environment variables.
+
+Let's assume you have following property defined in `application.properties`:
+
+[source,properties]
+----
+remote.host=quarkus.io
+----
+
+You can combine environment variables and property expressions by having a property defined as follows:
+
+[source,properties]
+----
+application.host=${HOST:${remote.host}}
+----
+
+This will expand the `HOST` environment variable and use the value of the property `remote.host` as the default value if `HOST` is not set.
+
+For the purpose of this section we used the property `remote.host` we defined previously.
+It has to be noted that the value could have been a fixed one such as in:
+
+[source,properties]
+----
+application.host=${HOST:localhost}
+----
+
+which will result in `localhost` as the default value if `HOST` is not set.
+
+[[configuring_quarkus]]
+== Configuring Quarkus
+
+Quarkus itself is configured via the same mechanism as your application. Quarkus reserves the `quarkus.` namespace
+for its own configuration. For example to configure the HTTP server port you can set `quarkus.http.port` in
+`application.properties`. All the Quarkus configuration properties are link:all-config[documented and searchable].
+
+[IMPORTANT]
+====
+As mentioned above, properties prefixed with `quarkus.` are effectively reserved for configuring Quarkus itself and
+therefore `quarkus.` should **never** be used as prefix for application specific properties.
+
+In the previous examples using `quarkus.message` instead of `greeting.message` would result in unexpected behavior.
+====
+
+Quarkus does much of its configuration and bootstrap at build time and most configuration properties are read and used during the build.
+These properties are _fixed at build time_ and it's not possible to change them at runtime.
+You always need to repackage your application in order to reflect changes of such properties.
+
+TIP: The properties fixed at build time are marked with a lock icon (icon:lock[]) in the link:all-config[list of all configuration options].
+
+However, some extensions do define properties _overridable at runtime_.
```

Comment:
```suggestion
However, some extensions do define properties that are _overridable at runtime_.
```

Reply:
Hm, I'll change this to _"However, some extensions do define properties that can be _overriden at runtime_."_

Reply:
:+1: